cmd/local-admin-client.go

```diff
@@ -74,6 +74,28 @@ func (lc localAdminClient) ServerInfo() (sid ServerInfoData, e error) {
 	}, nil
 }
 
+// Make a copy of the lockMap as the caller is going
+// to access lockMap without holding locks.
+func lockMapCopy(lockMap map[string][]lockRequesterInfo) map[string][]lockRequesterInfo {
+	lockCopy := map[string][]lockRequesterInfo{}
+	for k, v := range lockMap {
+		var lockSlice []lockRequesterInfo
+		for _, lockInfo := range v {
+			lockSlice = append(lockSlice, lockInfo)
+		}
+		lockCopy[k] = lockSlice
+	}
+	return lockCopy
+}
+
+// TopLocks - Returns the locks open on this server.
+func (lc localAdminClient) TopLocks() (TopLocksResp, error) {
+	if globalBootTime.IsZero() || globalLockServer == nil {
+		return TopLocksResp{}, errServerNotInitialized
+	}
+	return lockMapCopy(globalLockServer.ll.lockMap), nil
```

Comment:
I wonder how globalLockServer.ll.lockMap is not protected by a mutex since I guess it is always updated concurrently.

Reply:
Added code as suggested by @krishnasrinivas